parquet-column/src/main/java/org/apache/parquet/column/values/dictionary/DictionaryValuesWriter.java

```diff
@@ -285,8 +285,7 @@ public void fallBackDictionaryEncodedData(ValuesWriter writer) {
     }
 
     protected static Binary copy(Binary binary) {
-      return Binary.fromByteArray(
-          Arrays.copyOf(binary.getBytes(), binary.length()));
+      return Binary.fromByteArray(binary.getBytes());
```

Comment:
I think this should be built into `Binary`. There are times when `getBytes` will copy and it doesn't need to, like when the binary is backed by a byte array slice. It would be better if each Binary implementation had its own copy method.